programs/bubblegum/src/lib.rs

```diff
@@ -265,6 +265,36 @@ pub struct Compress<'info> {
     pub gummyroll_program: Program<'info, Gummyroll>,
 }
 
+pub fn hash_metadata(metadata: &MetadataArgs) -> Result<[u8; 32]> {
+    Ok(keccak::hashv(&[metadata.try_to_vec()?.as_slice()]).to_bytes())
+}
+
+pub enum InstructionName {
+    Unknown,
+    Mint,
+    Redeem,
+    CancelRedeem,
+    Transfer,
+    Delegate,
+    Decompress,
+}
+pub fn get_instruction_type(full_bytes: &Vec<u8>) -> InstructionName {
+    let disc: [u8; 8] = {
+        let mut disc = [0; 8];
+        disc.copy_from_slice(&full_bytes[..8]);
+        disc
+    };
+    match disc {
+        [51, 57, 225, 47, 182, 146, 137, 166] => InstructionName::Mint,
```

Comment:
I wonder if there's a better way to do this. There's probably a `discriminator` Trait that's implemented for each of the instructions

Reply:
fair point, will check this one more time before committing

Reply:
No there isn't a discriminator trait that does this